sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFramesSuite.scala

```diff
@@ -548,4 +548,19 @@ class DataFrameWindowFramesSuite extends QueryTest with SharedSparkSession {
       df,
       Row(1) :: Row(1) :: Nil)
   }
+
+  test("SPARK-45352: Remove window partition if partition expression are foldable") {
+    val ds = Seq((1, 1), (1, 2), (1, 3), (2, 1), (2, 2)).toDF("n", "i")
+    val sortOrder = SortOrder($"i".expr, Ascending)
+    val window1 = new WindowSpec(Seq(), Seq(sortOrder), UnspecifiedFrame)
+    val window2 = new WindowSpec(Seq(lit(1).expr), Seq(sortOrder), UnspecifiedFrame)
+    val df1 = ds.select(row_number().over(window1).alias("num"))
+    val df2 = ds.select(row_number().over(window2).alias("num"))
```

Comment:
for end-to-end tests, let's use end-to-end APIs, e.g. `Window.partitionBy($"key").orderBy($"value")`

Reply:
Updated, thanks.